src/Ssh/Ssh/SshCommands/EnterAzVMCommand.cs

```diff
@@ -142,6 +137,77 @@ private int StartSSHConnection()
             return sshProcess.ExitCode;
         }
 
+        private int StartRDPConnection()
+        {
+            Process sshProcess = null;
+            int success = 1;
+
+            try
+            {
+                // Get an open local port to act an a listener
+                TcpListener listener = new TcpListener(IPAddress.Loopback, 0);
+                listener.Start();
+                rdpLocalPort = ((IPEndPoint)listener.LocalEndpoint).Port;
+                listener.Stop();
+
+                // Start SSH Process
+                sshProcess = CreateSSHProcess();
+                sshProcess.StartInfo.RedirectStandardError = true;
+                sshProcess.Start();
+
+                bool sshSuccess = WaitSSHTunnelConnection(sshProcess);
```

Comment:
Should there be a timeout for this connection wait?  Can this wait be interrupted with Ctl+C?

Reply:
Yes, it can be interrupted with a Ctrl+c, and in that case the function returns false and the cmdlet exits. I'm not totally sure if having a timeout is necessary since this function exits once the ssh function exits, and ssh has its own timeout. But I'll look into it and make sure.